platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,1027 @@
+   reading the private `tssEncryptionKey`)
+2. The app will validate the cryptography and state.
+3. If a software upgrade is detected,
+   1. If the existing active roster and candidate roster do not have complete
+      key material and the network does not yet have a ledger id, the node will
+      adopt the candidate roster immediately on software upgrade.
+      - This will retain the pre-ledger-id behavior of the network until the
+        ledger id can be created for the first time.
+   2. If the candidate roster exists, has key material, and the number of yes
+      votes passes the threshold for adoption, then the candidate roster
+      will be adopted and become the new active roster.
+      1. Validation Check: if the state already has a ledger id set, the ledger
+         id recovered from the key material must match the ledger id
+         in the state.
+      2. If there is no ledger id in the state and the previously active roster
+         has no key material, the ledger id recovered from the candidate roster
+         key material will be set in the state as the new ledger id.
+   3. In all other cases the existing active roster will remain the
+      active roster.
+4. The application will provide the platform with the active roster to use.
+5. The platform will continue through its normal startup sequence.
+
+Note: If this is release N, then if all goes well and the candidate roster is successfully keyed, the next software
+upgrade (release N+1) is when the network will receive its ledger id. Release N+2 should carry forward the ledger
+id into the next roster automatically. No additional code deployment is required to make this happen.
+
+##### The Cleanup Release
+
+This release will contain any code cleanup required from the migration to the new TSS bootstrap process.
+
+### Component Architecture
+
+A new Service called `TSS Base Service` is introduced to the application layer. This service is responsible for all
+TSS related state and computation. In this proposal, the capability is scoped to keying the candidate rosters to
+generate a ledger id and transfer it from one roster to the next. In the `TSS-Block-Signing` proposal, the
+capability will be expanded to signing messages with share private keys and aggregating the signatures to create a
+ledger signature from the network.
+
+The TSS Base Service will be broken out into one interface and two components.
+
+1. `TssService` - the interface that defines the public API for the TSS Base Service.
+2. `TssStateManager` - executes on the transaction handling thread and updates state.
+3. `TssCryptographyManager` - executes on a separate thread and is responsible for cryptographic operations.
+4. Transaction Resubmission
+
+![TSS Platform Wiring](./TSS-Wiring-Diagram.drawio.svg)
+
+#### TssService
+
+The `TssService` interface provides the inter-service API for the TSS Base Service. It is responsible for setting the
```

Comment:
Just noting that there will be a name overlap with:
https://github.com/hashgraph/hedera-services/blob/develop/hedera-cryptography/hedera-cryptography-tss/src/main/java/com/hedera/cryptography/tss/api/TssService.java

Reply:
I can modify it to `TssBaseService` which is consistent with the naming that @rbair23 was using.